src/main/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamRestController.java

```diff
@@ -683,6 +683,49 @@ public TeamResource getTeamByName(HttpServletRequest request, Principal principa
 		return new ResponseEntity<>("User  request was successfully sent to team owner " + teamResource.getName(), HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/{teamName:.+}/contactemail/", method = RequestMethod.PUT)
+	public ResponseEntity<?> addTeamContactEmails(HttpServletRequest request, Principal principal,
+												  @PathVariable("teamName") String teamName,
+												  @RequestBody String emails) {
+		logger.info("User " + principal.getName() + " requested adding contacts to team " + teamName);
+
+		if(teamName == null || teamName.isEmpty()){
+			throw new TeamNameInvalidInputException("Team name should not be empty");
+		}
+
+		logger.info("Checking if user is team owner");
+		Team team = teamService.findByName(teamName);
+		if(!team.getAccount().getUsername().equals(principal.getName())){
+			throw new TeamNotFoundException(team.getName() + " is not found/not accessible by user");
+		}
+		Set<String> emailSet = Arrays.stream(emails.split(",")).collect(Collectors.toSet());
+		team = teamService.setContactEmails(emailSet, team);
+		return new ResponseEntity<>("Team Contact was added to team " + team.getName(), HttpStatus.OK);
```

Comment:
This is not a good response for object creation in REST. Normally you return an object representation for PUT.

Reply:
For POST it is required, for PUT isn't status enough?